packages/ethereum/contracts/HoprDistributor.sol

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./HoprToken.sol";
+import "./utils/console.sol";
+
+contract HoprDistributor is Ownable {
+    using SafeMath for uint256;
+
+    // helps us create more accurate calculations
+    uint32 public constant MULTIPLIER = 10 ** 6;
+
+    // total amount minted
+    uint128 public totalMinted = 0;
+
+    // time where the contract will consider as starting time
+    uint32 public startTime;
+    // token which will be used
+    HoprToken public token;
+    // maximum tokens allowed to be minted
+    uint128 public maxMintAmount;
+
+    // A {Schedule} that defined when and how much will be claimed
+    // from an {Allocation}
+    struct Schedule {
+        uint32[] durations;
+        uint32[] percents;
+    }
+
+    // An {Allocation} represents how much a account can claim, claimed,
+    // and when last claim occured
+    struct Allocation {
+        uint128 amount;
+        uint128 claimed;
+        uint32 lastClaim;
+        bool revoked; // account can no longer claim
+    }
+
+    // schedule name -> Schedule
+    mapping(string => Schedule) private schedules;
+
+    // account -> schedule name -> Allocation
+    // allows for an account to have more than one type of Schedule
+    mapping(address => mapping(string => Allocation)) private allocations;
+
+    /**
+     * @param _startTime The timestamp to start counting
+     * @param _token The token which we will mint
+     */
+    constructor(HoprToken _token, uint32 _startTime, uint128 _maxMintAmount) public {
+        startTime = _startTime;
+        token = _token;
+        maxMintAmount = _maxMintAmount;
+    }
+
+    /**
+     * @dev Revokes the ability for an account to claim on the
+     * specified schedule.
+     * @param account the account to crevoke
+     * @param scheduleName the schedule name
+     */
+    function revokeAccount(
+        address account,
+        string calldata scheduleName
+    ) external onlyOwner {
+        require(allocations[account][scheduleName].amount != 0, "Allocation must exist");
```

Comment:
here we should loop over the schedule and remove all remaining allocation parts from the `totalToBeMinted`